pkg/plugins/common/kustomize/v1/create.go

```diff
@@ -35,7 +35,10 @@ type createSubcommand struct {
 	force bool
 }
 
-func (p *createSubcommand) BindFlags(fs *pflag.FlagSet) { p.flagSet = fs }
+func (p *createSubcommand) BindFlags(fs *pflag.FlagSet) {
+	p.flagSet = fs
+	_ = fs.MarkDeprecated("component-config", "component-config flag will be deprecated in upcoming releases")
```

Comment:
But we are deprecating when we use `MarkDeprecated`.
That will result in for example this option no longer be doc in sdk docs when we run make generate to generate the docs.

Reply:
/hold